doc/developer/design/20230225_compute_introspection_schema.md

```diff
@@ -0,0 +1,217 @@
+GROUP BY operator_id, worker_id"
+```
+
+This dimension is applied inconsistently: not all relations that
+report a count or sum in the data are derived from a corresponding
+`_internal` source. 
+
+`raw` variants, on the other hand, measure some time value in
+nanoseconds, whereas their corresponding non-`raw` views convert the
+times to the SQL interval type.
+
+Confusingly, not all times are reported as intervals; for example,
+`mz_scheduling_parks` has `slept_for` and `requested` columns, whose
+type is `bigint` and whose unit is undocumented (in fact, it is nanoseconds).
+
+#### Inconsistent presence of `sum` column in histograms
+
+Some histograms include a per-bucket sum, which allows consumers to
+compute average or total values precisely. Confusingly, not all
+histograms report such a column (e.g. `mz_worker_compute_delays` does
+not, whereas `mz_worker_raw_compute_delays` does).
+
+#### Lack of clarity around per-replica vs. global relations
+
+Many of the relations are logging sources that run locally in
+clusters, are not persisted, and report different data depending on
+the currently selected replica. Views based on these sources are also
+per-replica. It is not straightforward to tell from our docs which is
+the case for each relation.
+
+Furthermore, sometimes the same logical value is reported by two
+relations, one reporting `clusterd`'s view of the world, and the other
+reporting `environmentd`'s. For example, `mz_compute_frontiers`
+vs. `mz_cluster_replica_frontiers`. However, the names do not make it
+clear what the distinction is. 
+
+### Suggested fixes for schema inconsistencies
+
+* All per-worker relations should have a corresponding "global" view
+  where the results are aggregated across workers in the way most
+  likely to be useful (for example, summing, taking the minimum, or
+  restricting to worker 0). Because these global views are more likely
+  to be useful than the underlying sources, they should have the basic
+  name (e.g., `mz_arrangement_sharing`). The underlying source should
+  have a name with `per_worker` suffixed (e.g.,
+  `mz_arrangement_sharing_per_worker`). Furthermore, the per-worker
+  variant should be clearly less prominent in documentation (either at
+  the bottom of the page, or in a "see more" section that has to be
+  clicked to expand).
+* The expectations about per-worker relations should be clearly
+  documented (e.g., we should clearly say whether the values for each
+  worker are expected to be the same during normal operation).
+* All relations that report a simple count or sum should be views based on
+  an `_internal` variant that maintains the count or sum in the
+  Differential diff field. We may consider renaming this, since it
+  conflicts confusingly with the overall name `mz_internal`.
+* `raw` variants should be removed, and the underlying dataflows
+  should be in terms of intervals.
+* All histogram views or sources should have a per-bucket `sum`
+  column.
```

Comment:
Does that imply that we won't be able to use DD's diff field anymore for maintaining the bucket counts? I think that roughly doubles our traffic on those collections, because we now need to retract previous values, which we didn't have to before. Probably still fine, since the number of buckets limits how many changes we can see per time interval.

Reply:
The DD diff field can be a two-element `(sum, count)` tuple, so in principle you could still use it. When I implemented bucket counts for `frontier_delay` we went back and forth on this a few times.

Reply:
But when we want to sink to persist, our diff type can only be `i64`, right?

Reply:
You're right, it has to go to `i64` at any boundary with the outside world. I think it's okay, the traffic in these tables should be rather small.

Reply:
s/tables/sources/